### ISSUE TITLE
Avoid generating machine-id during runtime

### ISSUE THREAD

**Issue by K900** [scaleway/image-tools#181]
As noted [here](https://github.com/systemd/systemd/issues/4025#issuecomment-242164964), that behavior is explicitly unsupported, and can lead to weird issues like the one originally described in the ticket.

**Comment by QuentinPerez** [comment]
Hi @K900,

Thank you for your feedback, I will check if I can find another way.

**Comment by K900** [comment]
I think the "right" way of doing things is to generate it at provisioning time, and then add `systemd.machine_id=[whatever]` to the kernel command line in the bootscript. The official [manual page](https://www.freedesktop.org/software/systemd/man/machine-id.html) seems to suggest the same thing. All the other ways of setting it require another running system with access to the new rootfs, so that's not going to work assuming I understand the deployment process correctly.

**Comment by rdorsch** [comment]
A side effect of this is that all Debian jessie upgrades to stretch break on scaleway and need a manual change of /etc/machine-id:

The user needs to remove the second line of

```
root@scw-790923:~# cat /etc/machine-id 
9d1b906dd5ea40359e2071d29c12aabe
71f
root@scw-790923:~# 
```

 manually before the upgrade.

Otherwise he will see messages like

`logger: socket /dev/log: Connection refused`

and many packages do not upgrade properly.

I hope scaleway can handle this issue properly and not all users on scaleway break their systems after Debian Stretch is released.

**Comment by bchatelard** [comment]
Fixed in a0a2e37adde958200d5db089c0f3517de502bf45

Related to scaleway/image-debian#71
